ci/scripts/integration_arrow.sh

```diff
@@ -34,11 +48,11 @@ time archery integration \
     --run-c-data \
     --run-ipc \
     --run-flight \
-    --with-cpp=1 \
-    --with-csharp=1 \
-    --with-java=1 \
-    --with-js=1 \
-    --with-go=1 \
+    --with-cpp=$([ "$ARROW_INTEGRATION_CPP" == "ON" ] && echo "1" || echo "0") \
+    --with-csharp=$([ "$ARROW_INTEGRATION_CSHARP" == "ON" ] && echo "1" || echo "0") \
+    --with-go=$([ "$ARROW_INTEGRATION_GO" == "ON" ] && echo "1" || echo "0") \
+    --with-java=$([ "$ARROW_INTEGRATION_JAVA" == "ON" ] && echo "1" || echo "0") \
+    --with-js=$([ "$ARROW_INTEGRATION_JS" == "ON" ] && echo "1" || echo "0") \
```

Comment:
I think we need a with-rust here in order to use this in arrow-rs

Reply:
There's no with-rust currently. The `ARCHERY_INTEGRATION_WITH_RUST` environment variable is used.

Reply:
We're currently using it - https://github.com/apache/arrow-rs/blob/master/.github/workflows/integration.yml#L121

I'll give the env variable a go as well

Reply:
Ahah... I don't know why we have both then.